MantidQt/CustomInterfaces/src/Reflectometry/ReflRunsTabPresenter.cpp

```diff
@@ -278,12 +290,10 @@ void ReflRunsTabPresenter::transfer() {
     return;
   }
 
-  for (auto rowIt = selectedRows.begin(); rowIt != selectedRows.end();
-       ++rowIt) {
-    const int row = *rowIt;
-    const std::string run = m_searchModel->data(m_searchModel->index(row, 0))
-                                .toString()
-                                .toStdString();
+  for (auto &&row : selectedRows) {
```

Comment:
Interesting reads [here](https://edmundv.home.xs4all.nl/blog/2014/01/28/use-auto-and-and-for-range-based-for-loops/) and [here](https://blog.petrzemek.net/2016/08/17/auto-type-deduction-in-range-based-for-loops/).

To summarise `auto&&` is a good default for generic code since it works all the time. But for non generic code `const auto&` can be considered better since it can be unnecessarily confusing.